lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -102,14 +102,19 @@ def compute_value_lock(self, value_key: str) -> threading.Lock:
         with self._value_locks_lock:
             return self._value_locks[value_key]
 
-    def clear(self):
-        """Clear all values from this cache."""
+    def clear(self, key: str | None = None):
+        """Clear values from this cache.
+        If no argument is passed, all items are cleared from the cache.
+        A key can be passed to clear that key from the cache only."""
```

Comment:
cc @sfc-gh-dmatthews to double-check the docstring text. But docstring changes will probably go into a separate PR or?

Reply:
I can go either way. If you want to get this merged right away, feel free to. Otherwise, I can try to take a look at this tomorrow or Monday. (Just going through tasks after coming back from a few days off....)

Reply:
I think it is easier to merge this right now since the code is ready and do the docs iteration in a dedicated PR. Btw. this wrong code usage within the docstring example would also need to be fixed: https://github.com/streamlit/streamlit/pull/8297#discussion_r1571010599